packages/ai/src/ui/chat-store.ts

```diff
@@ -0,0 +1,518 @@
+import { parsePartialJson } from '../util';
+import type {
+  ReasoningUIPart,
+  TextUIPart,
+  ToolInvocation,
+  ToolInvocationUIPart,
+  UIMessage,
+} from './ui-messages';
+
+interface ChatStoreSubscriber {
+  id: string;
+  onChatChanged: (event: ChatStoreEvent) => void;
+}
```

Comment:
I would prefer generic subscribers (not specific to chat `id`) and then push the id filtering into the hook (useChat) -- this will enable more flexible clients to chatStore.